### PR TITLE
Better False-Positive Detection in rz_scan_strings

### LINKED ISSUE #1836: Fix UTF-64LE string detection when contains `\xa1` character
```
[XX] db/cmd/metadata Csa, Cs. and Cs.l
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'e str.escbslash=true
s 0x140016018
Csa
Csl*~`spad`
Cs.q
Cs.
Cs.l
pd 1
Cs-
Csw
Csl*~`spad`
Cs.q
Cs.
Cs.l
pd 1
Cs-
Csa 4
Cs.l
Cs.l @ 0x14001601c  # should print nothing
Csa 4
Cs.l
Cs.l @ 0x14001601c  # should print nothing
' bins/pe/testapp-msvc64.exe
-- stdout
--- expected
+++ actual
@@ -4,10 +4,11 @@
 0x140016018 ascii[2] "\t"
             ;-- str.wide_esc:___0m:
             0x140016018     .string "\t" ; len=2
-Csw 19 @ 0x140016018 # \twide\\esc: \e[0m\xa1\r\n
-"\twide\\esc: \e[0m\xa1\r\n"
-ut16le[15] "\twide\\esc: \e[0m\xa1\r\n"
+Csw 31 @ 0x140016018 # \twide\\esc: \e[0m
+"\twide\\esc: \e[0m"
+utf16le[31] "\twide\\esc: \e[0m"
+0x140016018 utf16le[31] "\twide\\esc: \e[0m"
             ;-- str.wide_esc:___0m:
-            0x140016018     .string "\twide\\esc: \e[0m\xa1\r\n" ; len=19
+            0x140016018     .string "\twide\\esc: \e[0m" ; len=31
 0x140016018 ascii[4] "\t"
 0x140016018 ascii[4] "\t"
```

See the test called `Csa, Cs. and Cs.l` in `test/db/cmd/metadata.

Regression was introduced in https://github.com/rizinorg/rizin/commit/39613b5c16401544c9841ba96d9fafff36638315 (https://github.com/rizinorg/rizin/pull/1752/)

Fix is likely required in these files:
- `librz/util/str_search.c`
- `libz/util/utf8.c`
- `librz/util/str.c`

### PR DESCRIPTION
**Pull request by borzacchiello**
Hello everyone!
I tried to improve rz_scan_strings, with the main goal of making the false-positive check more robust, and fix #1836.
I also added a global option to activate/deactivate a check in the false-positive detection function. I am not sure that I added it in the correct way, so let me know.

 

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This PR adds the following features:
- Extend the false-positive check on ASCII frequencies to all UTF strings (previously it was performed only on UTF16LE strings).
- Add a global option to activate/deactivate the above-mentioned check.
- Improve the false-positive heuristic by adding a special case for extended-ASCII strings. This is useful for European languages.

I had to modify some regression test because some of them checked the number of detected strings that is changed due to the PR.
The removed strings should be only false positives: this [zip file](https://github.com/rizinorg/rizin/files/8893588/strings_diff.zip) contains a diff for each binary of the regression tests for which the number of strings found changed.

**Test plan**

...

**Closing issues**

closes #1836

### CODE REVIEW

**Review by wargio — APPROVED** [review]

File: librz/core/cconfig.c [hunk 1]
```diff
@@ -3588,6 +3595,7 @@ RZ_API int rz_core_config_init(RzCore *core) {
 
 	/* str */
 	SETCB("str.escbslash", "false", &cb_str_escbslash, "Escape the backslash");
+	SETCB("str.search.check_ascii_freq", "true", &cb_strsearch_check_ascii_freq, "Skip ASCII frequency analysis when looking for false positives during string search");
```
Comment (XVilka) [inline librz/core/cconfig.c]: Split the long line

**Review by XVilka — CHANGES_REQUESTED** [review]
Looks good! I will check more with non-European languages tonight and will merge if it works well. Awesome job! 💯

File: librz/core/cconfig.c (hunk 1 above)
Comment (borzacchiello) [reply librz/core/cconfig.c]: done!

**Review by XVilka — APPROVED** [review]
Does not cause regressions on non-European languages. 👍

**Pull request merged**